### PR TITLE
Subnautica: updates

### PR DESCRIPTION
**Pull request by Berserker66**
Adds new goals; drive core repair, maximum infection and disable quarantine.
Forces two Seaglide fragments into local sphere 1, mimicing the base game as playing without seaglide is a chore and not fun.
Fixes a databox position which lead to it never sending on the client.

### CODE REVIEW

File: worlds/subnautica/__init__.py [hunk 1]
```diff
@@ -34,12 +32,30 @@ class SubnauticaWorld(World):
...
+
+    def create_regions(self):
+        self.world.regions += [
+            self.create_region('Menu', None, ['Lifepod 5']),
+            self.create_region('Planet 4546B',
+                               Locations.events + [location["name"] for location in Locations.location_table.values()])
+        ]
+
+    def set_rules(self):
+        set_rules(self.world, self.player)
```
Comment (black-sliver) [inline worlds/subnautica/__init__.py:58]: this is confusing to a reader
maybe `from . import Rules; Rules.set_rules(self.world, self.player)` would be better?

File: worlds/subnautica/__init__.py [hunk 2]
```diff
@@ -72,44 +87,62 @@ def generate_basic(self):
...
-    ret.world = world
-    if locations:
-        for location in locations:
-            loc_id = locations_lookup_name_to_id.get(location, 0)
-            location = SubnauticaLocation(player, location, loc_id, ret)
-            ret.locations.append(location)
-    if exits:
-        for exit in exits:
-            ret.exits.append(Entrance(player, exit, ret))
+        item_id: int = self.item_name_to_id[name]
```
Comment (black-sliver) [inline worlds/subnautica/__init__.py:113]: wouldn't it be better to add type info to the dict?

**Review by black-sliver — APPROVED** [review]
Code looks good, rolling works, left two comments. I did not actually attempt to play a seed though.

File: worlds/subnautica/__init__.py (hunk 2 above)
Comment (Berserker66) [reply worlds/subnautica/__init__.py:113]: type info is in the dict as well.

Comment (black-sliver) [reply worlds/subnautica/__init__.py:113]: my point is that you don't have to add type hints to local variables if they get assigned from a fully typed source, but it doesn't hurt either

**Review by black-sliver — APPROVED** [review]
Still approved :-)

**Pull request merged**